gson/src/test/java/com/google/gson/functional/JsonParserTest.java

```diff
@@ -112,11 +104,8 @@ public void testBadFieldTypeForDeserializingCustomTree() {
     obj.add("primitive1", primitive1);
     obj.add("primitive2", array);
 
-    try {
-      gson.fromJson(obj, Nested.class);
-      fail("Nested has field BagOfPrimitives which is not an array");
-    } catch (JsonParseException expected) {
-    }
+    // Nested has field BagOfPrimitives which is not an array
```

Comment:
The test was previously only calling `fail(...)` with that message. So if the test unexpectedly did not throw an exception, it would have failed with that message.

But it did not perform any assertions on the message of the expected thrown exception.

Or am I misunderstanding you?

Reply:
Dunno what I was thinking.